### PR TITLE
add a frogger game

### PR DESCRIPTION
**Pull request by dafen173**
### OOP exercise - Frogger game

[Demo](https://dafen173.github.io/frogger-game/)
[Code base](https://github.com/dafen173/frogger-game/blob/master/js/app.js)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@dafen173 demo is 404.
Check the PR template (you actually didn't change it) for a link to publishing instructions.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
Please, confirm all items below are positive, having the code fixed as appropriate. Please, ask a question if anything below requires clarification.

Minimal requirements to meet:
 * [ ] it is OK to employ ES6 features like `const`, `let` etc.
 * [ ] the code is very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
 * [ ] OO is implemented using JS prototype chain object model (**not** ES6 classes syntax)
 * Requirements re **Constants**:
   * [ ] all numbers like block dimensions, initial locations are defined as named constants
         (e.g. `const STEP = 101;`) as otherwise numbers scattered across code base look cryptic;
         named constants add semantic meaning and improve readability
   * [ ] every number that have a semantic purpose (like those listed above) should be defined as constants;
         think of how your code **reads** - the closer to plain English the better
   * [ ] there are core constants and derived constants
         (e.g. derived constant `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
   * [ ] arrays of constants are also constants 
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
   * [ ] const objects help organizing and structure const data even better
         (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`
 * Requirements re **OOP**:
   * [ ] classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class
         (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
         Hint: pass `Player` instance as an argument to every enemy
   * [ ] Separation of Concerns principle is followed
         (e.g. `update` method does only rendering and doesn't contain any unrelated **inline** code;
         for example collision check is defined as a dedicated method and only called from inside `update`)
   * [ ] Nice To Have: properties common for some classes are generalized into a base class
         (e.g. there is `Character` base class, which is extended by `Enemy` and `Player` classes)
   * [ ] class extension is implemented using `Subclass.prototype = Object.create(Superclass.prototype)`,
         not `Subclass.prototype = new Superclass(params);`;
         [Useful resource](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
 * Most common mistakes
   * [ ] Make sure `target = condition ? valueWhenConditionTrue : valueWhenConditionFalse` is used instead of
         `condition ? target = valueWhenConditionTrue : target = valueWhenConditionFalse`;
         [Conditional (ternary) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)

**Comment by dafen173** [comment]
> 
> 
> @dafen173 demo is 404.
> Check the PR template (you actually didn't change it) for a link to publishing instructions.

Fixed. So ok?

**Comment by dafen173** [comment]
Fixed. Please, review!

File: submissions/dafen173/frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+	checkCollisions () {
+		if (player.x < this.x + ENEMY_CONF.WIDTH &&
+        	player.x + ENEMY_CONF.WIDTH > this.x &&
+        	player.y < this.y + ENEMY_CONF.HEIGHT &&
+        	player.y + ENEMY_CONF.HEIGHT > this.y) {
+        		player.x = PLAYER_CONF.START_X;
+        		player.y = PLAYER_CONF.START_Y;
+    		}
+	}
```
Comment (OleksiyRudenko) [inline submissions/dafen173/frogger-game/app.js:90]: One of the checklist items says:
```
classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class 
(referring to global constants and globals provided by the gaming platform like `Resources` is OK);
Hint: pass `Player` instance as an argument to every enemy
```

Please check the entire checklist against your entire codebase carefully.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@dafen173 nearly done

**Comment by OleksiyRudenko** [comment]
@dafen173 Please let us know until 20:00 today if you are going to submit any changes into this PR before end of the day February 27. This would help us to plan mentors' limited capacity accordingly.

**Comment by OleksiyRudenko** [comment]
Feel free to reopen this PR once you feel ready to submit any updates.

**Pull request closed without merging**